### PR TITLE
Enabled warnings as errors

### PR DESCRIPTION
**Pull request by shamsasari**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -242,6 +242,12 @@ allprojects {
             exclude module: 'isolated'
         }
     }
+
+    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+        kotlinOptions {
+            allWarningsAsErrors = true
```
Comment (chrisr3) [inline build.gradle]: Don't we already have a `KotlinCompile` block? Why not add the `allWarningsAsError=true` line to that instead?

File: core/src/main/kotlin/net/corda/core/transactions/WireTransaction.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ class WireTransaction(componentGroups: List<ComponentGroup>, val privacySalt: Pr
     @DeleteForDJVM
     constructor(componentGroups: List<ComponentGroup>) : this(componentGroups, PrivacySalt())
 
-    @Deprecated("Required only in some unit-tests and for backwards compatibility purposes.", ReplaceWith("WireTransaction(val componentGroups: List<ComponentGroup>, override val privacySalt: PrivacySalt)"), DeprecationLevel.WARNING)
+    @Deprecated("Required only for backwards compatibility purposes.", ReplaceWith("WireTransaction(val componentGroups: List<ComponentGroup>, override val privacySalt: PrivacySalt)"), DeprecationLevel.WARNING)
```
Comment (chrisr3) [inline core/src/main/kotlin/net/corda/core/transactions/WireTransaction.kt]: This is actually used by `EnclaveletTest`.

File: build.gradle (hunk 1 above)
Comment (shamsasari) [reply build.gradle]: Good spot.

**Comment by shamsasari** [comment]
Successor to https://github.com/corda/corda/pull/2516

**Pull request merged**